cmd/generate/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"text/template"
+
+	"github.com/rudderlabs/rudder-observability-kit/cmd/internal/generator"
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed labels.yaml
+var labelsYAML []byte
+
+func generateLabels(labels generator.Labels, templateFile string, outputDir string, extension string) {
+	tmpl, err := template.New(path.Base(templateFile)).ParseFiles(templateFile)
+	if err != nil {
+		log.Fatal("error occurred while parsing template", err)
+	}
+	for team, teamLabels := range labels.Labels {
+		fileName := fmt.Sprintf("%s/%s.%s", outputDir, team, extension)
+		file, err := os.Create(fileName)
+		if err != nil {
+			log.Fatal("error occurred while creating file", err)
+		}
+		defer file.Close()
```

Comment:
Can we move the for loop content into a function to avoid the linting on the "don't call defer in a for loop"?
The argument is that the defers would pile up and we would have one defer per element in labels and they would all run at the end. If we wrap it in a function we would run one defer at a time.